npc/re/merchants/hd_refiner.txt

```diff
@@ -274,38 +336,46 @@ OnInit:
 		mes "I really am the best blacksmith in the whole wide world!";
 		close;
 	}
-	downrefitem .@part;
-	next;
-	emotion (!rand(5))?e_cash:e_omg;
-	mes "[Basta]";
-	mes "Aaaaaaaaaaak!!!";
-	next;
-	mes "[Basta]";
-	mes "Damn it!";
-	mes "Refining failed and refine level has decreased!";
-	mes "Even the best blacksmith in the world doesn't guarantee 100% success!";
+	if (.@bless_who == 1) {
+		specialeffect EF_HOLYHIT;
+		next;
+		emotion (rand(5) ? e_omg : e_cash );
+		mes "[Basta]";
+		mes "Aaaaaaaaaaak!!!";
+		next;
+		mes "[Basta]";
+		mes "Refining is failed!";
+		mes "The best blacksmith in the world like me...";
+		mes "doesn't guarantee 100% success~";
+	}
+	else {
+		downrefitem .@part;
+		next;
+		emotion (rand(5) ? e_omg : e_cash );
+		mes "[Basta]";
+		mes "Aaaaaaaaaaak!!!";
+		next;
+		mes "[Basta]";
+		mes "Damn it!";
+		mes "Refining failed and refine level has decreased!";
+		mes "Even the best blacksmith in the world doesn't guarantee 100% success!";
+	}
 	mes "Too bad.";
 	next;
 	mes "[Basta]";
 	mes "I'll do better next time! Don't worry!";
 	close;
-}
-prt_in,57,54,3	duplicate(Basta)	Basta#prt	826
-morocc_in,68,30,3	duplicate(Basta)	Basta#morocc	826
-payon,148,174,3	duplicate(Basta)	Basta#payon	826
-alberta_in,18,56,3	duplicate(Basta)	Basta#alberta	826
-yuno_in01,173,18,3	duplicate(Basta)	Basta#yuno	826
-ein_in01,24,82,3	duplicate(Basta)	Basta#einbroch	826
-lhz_in02,280,17,3	duplicate(Basta)	Basta#lighthalzen	826
-
-// Refine UI makes these NPCs useless
--	script	RefineUIBastaInit	-1,{
-	end;
 
-OnInit:
-	if( getbattleflag( "feature.refineui" ) == 3 ){
-		unloadnpc "Basta";
-	}
+OnInit:	// Refine UI makes these NPCs useless
+	if (getbattleflag("feature.refineui") == 3)
+		unloadnpc strnpcinfo(0);
```

Comment:
`unloadnpc strnpcinfo(3);`

Reply:
Map server crashes when changing the value to strnpcinfo(3); I have applied cyd's patch for refine. Will it have conflict?

Using strnpcinfo(0); map-server is normal.